config.yaml.dist

```diff
@@ -43,119 +42,105 @@ provider:
       # authentication documentation for further details.
-      username_claim: "sub"
-
+      usernameClaim: "sub"
       # If provided, all usernames will be prefixed with this value. If not
       # provided, username claims other than 'email' are prefixed by the issuer
       # URL to avoid clashes. To skip any prefixing, provide the value '-'.
-      username_prefix: "oidc:"
-
+      usernamePrefix: "oidc:"
       # If provided, the name of a custom OpenID Connect claim for specifying
       # user groups. The claim value is expected to be a string or array of
       # strings. This flag is experimental in kubernetes, please see the
       # kubernetes authentication documentation for further details.
-      groups_claim: "groups"
-
+      groupsClaim: "groups"
       # If provided, all groups will be prefixed with this value to prevent
       # conflicts with other authentication strategies.
-      groups_prefix: "oidc:"
-
+      groupsPrefix: "oidc:"
       # Comma-separated list of allowed JOSE asymmetric signing algorithms. JWTs
       # with a 'alg' header value not in this list will be rejected. Values are
       # defined by RFC 7518 https://tools.ietf.org/html/rfc7518#section-3.1.
-      signing_algs: "RS256"
-
+      signingAlgs: "RS256"
       # A key=value pair that describes a required claim in the ID Token. If
       # set, the claim is verified to be present in the ID Token with a matching
       # value. Only single pair is currently supported.
-      required_claim: ""
-
+      requiredClaim: ""
       # If set, the OpenID server's certificate will be verified by one of the
       # authorities in the oidc-ca-file, otherwise the host's root CA set will
       # be used.
-      ca_file: ""
+      caFile: ""
 
 # The list of nodes can be overwritten by providing Terraform output.
 # You are strongly encouraged to provide an odd number of nodes and
 # have at least three of them.
 # Remember to only specify your *master* nodes.
 # hosts:
-# - public_address: '1.2.3.4'
-#   private_address: '172.18.0.1'
-#   ssh_port: 22 # can be left out if using the default (22)
-#   ssh_username: ubuntu
-
-#   # Uou usually want to configure either a private key OR an
+# - publicAddress: '1.2.3.4'
+#   privateAddress: '172.18.0.1'
+#   sshPort: 22 # can be left out if using the default (22)
+#   sshUsername: ubuntu
+#   # You usually want to configure either a private key OR an
 #   # agent socket, but never both. The socket value can be
 #   # prefixed with "env:" to refer to an environment variable.
-#   ssh_private_key_file: '/home/me/.ssh/id_rsa'
-#   ssh_agent_socket: 'env:SSH_AUTH_SOCK'
+#   sshPrivateKeyFile: '/home/me/.ssh/id_rsa'
+#   sshAgentSocket: 'env:SSH_AUTH_SOCK'
 
 # The API server can also be overwritten by Terraform. Provide the
 # external address of your load balancer or the public address of
 # your first node.
-# apiserver:
-#   address: '1.2.3.4'
+# apiEndpoints:
```

Comment:
why is this an array if endpoint is a single point. And in kubeadm api it's also [a single string](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go#L83)

Reply:
That was some initial effort to try to adopt Cluster-API or align with it. I decided not to do that, but `apiEndpoints` change was left on purpose.

The decision to rename it is because `apiEndpoints` is more descriptive than `apiserver`. It tells the user for what exactly is it used for.

As of why is this an array—is it always going to be a single point? For example, what if we don't have a load balancer? Can the user access all the nodes or just the first one? I believe that depends on the setup. For users using KubeOne with KubeOne CLI, that mostly likes means nothing, but if someone is using KubeOne as a Go library it could be helpful.

Reply:
>is it always going to be a single point

yes, that's how [kubeconfig is arranged](https://github.com/kubernetes/client-go/blob/master/rest/config.go#L57) and the same is kubeadm.

Reply:
After discussion on the daily meeting, we decided to use only one API endpoint. Pushed a commit that fixes this.